exercise_1/Graph.hpp

```diff
@@ -0,0 +1,37 @@
+//
+// Created by denys on 08.07.19.
+//
+
+#pragma once
+
+#include <list>
+#include <fstream>
+#include <regex>
+#include <map>
+#include "Node.hpp"
+
+
+class Graph {
+private:
+    std::list<Node *>               _list_nodes;
+    std::map<std::string, Node *>   _map_nodes;
+
+    Node                *_create_or_get_node(const std::string &node_name);
+    void                _add_nodes(const std::string &name_first, const std::string &name_second, int weight);
+    Node                *_get_node(const std::string &node_name);
+    void                _start_dijkstras_algorithm();
+    Node                *_find_min_weight_node(std::list<Node *> &nodes);
+    static void _print_path(Node *start_point, Node *end_point);
+
+public:
+    Graph(std::istream &istream);
+    ~Graph();
```

Comment:
= default.

Reply:
As I know, default key word mean that default constructor should be used, but I have written my own constructor. Maybe you mean I should write " = explicit "?

Reply:
My comment is about destructor.

Reply:
A, I should do that after I will use smart pointers?

Reply:
Exactly. Smart pointers free the memory automatically. You do not have to bother with it.

Reply:
OK, thanks

Reply:
It's hard for now, without clear understanding of smart pointers, rewrite existing logic. Maybe it would be better to start making third task with smart pointers, and then after they will be familiar to me, I will rewrite existing code?

Reply:
OK - fair enough.

Reply:
Hi, I've decided to use advice from Stanisław regarding smart pointers. So I'm just getting the pointers from std::map. Could it be like that?

Reply:
I'm not sure if I understand you correctly. Please deliver some code, then I will check.

Reply:
The code is available: "exercise_1 Fix github comments" (5384823)